core/shared/src/test/scala/fs2/StreamSpec.scala

```diff
@@ -292,19 +292,22 @@ class StreamSpec extends Fs2Spec {
         }
       }
 
-      "interruption" in forAll { (s0: Stream[Pure, Int]) =>
-        Counter[IO].flatMap { counter =>
-          var ecs: Chain[ExitCase[Throwable]] = Chain.empty
-          val s =
-            Stream
-              .bracketCase(counter.increment) { (_, ec) =>
-                counter.decrement >> IO { ecs = ecs :+ ec }
-              }
-              .flatMap(_ => s0 ++ Stream.never[IO])
-          s.interruptAfter(50.millis).compile.drain.flatMap(_ => counter.get).asserting { count =>
-            count shouldBe 0L
-            ecs.toList.foreach(_ shouldBe ExitCase.Canceled)
-            Succeeded
+      "interruption" in {
+        pending // Completes with ExitCase.Completed instead of expected Canceled
```

Comment:
This is currently broken, but I think it's for the same reason as #1531 and recent reports in Gitter -- https://gitter.im/functional-streams-for-scala/fs2?at=5d644fb6022dba538e66cdfe.

I think when there's more granular scopes, the interruption applies to innermost scope but then the parents are terminated w/ `Completed`. I spent an hour or so looking in to this but couldn't figure it out. I think the issue is in `Algebra#scope0`, where the handler for interruption closes the interrupted scope but then terminates with a `Result.Pure(())`.

Reply:
I remember these was really hard to solve and reason about and took really long time to figure them out. Some of the changes on scope was purposely driven by these tests.  I hope they won't block us now.

Reply:
I agree, though I don't think the status quo is much better, assuming it's the cause of the other reported issues with `ExitCase.Completed` appearing where `ExitCase.Canceled` is expected.

Reply:
Okay fixed, it was very simple issue: 04b893ebba9d17144b99b759907ad5fef8a9bb0c